<launch>
    <!-- Drone id -->
    <arg name="drone_id"/>
    <arg name="drone_id_ns" value="drone$(arg drone_id)"/>
    <param name="/use_sim_time" value="false" />

    <group ns="$(arg drone_id_ns)">
      <arg name="fcu_url" value="/dev/ttyAML1:230400" />
      <!-- GCS: laptop-->
      <arg name="gcs_url" value="udp://@192.168.31.22" />
      <!-- GCS: Vicon Desktop-->
      <!-- <arg name="gcs_url" value="udp://@192.168.31.61" /> -->

      <!-- MAVROS -->
      <node pkg="mavros" type="mavros_node" name="mavros" required="true" clear_params="true" output="screen">
        <param name="fcu_url" value="$(arg fcu_url)" />
        <param name="gcs_url" value="$(arg gcs_url)" />
        <param name="target_system_id" value="$(eval 1 + arg('drone_id'))"/>
        <param name="target_component_id" value="1" />
        <param name="fcu_protocol" value="v2.0" />

        <!-- load blacklist, config -->
        <rosparam command="load" file="$(find gestelt_bringup)/config/px4_pluginlists.yaml" />
        <rosparam command="load" file="$(find gestelt_bringup)/config/px4_config.yaml" />

        <!-- <param name="global_position/frame_id" value="$(arg drone_id_ns)_origin"/>
        <param name="global_position/child_frame_id" value="$(arg drone_id_ns)_base_link"/>
        <param name="global_position/tf/frame_id" value="$(arg drone_id_ns)_origin"/>
        <param name="global_position/tf/global_frame_id" value="world"/>
        <param name="global_position/tf/child_frame_id" value="$(arg drone_id_ns)_base_link"/>

        <param name="local_position/frame_id" value="$(arg drone_id_ns)_origin"/>
        <param name="local_position/tf/send" value="true"/>
        <param name="local_position/tf/frame_id" value="$(arg drone_id_ns)_origin"/>
        <param name="local_position/tf/child_frame_id" value="$(arg drone_id_ns)_base_link"/> -->

        <param name="local_position/frame_id" value="world"/>
        <param name="local_position/tf/send" value="true"/>
        <param name="local_position/tf/frame_id" value="world"/>
        <param name="local_position/tf/child_frame_id" value="$(arg drone_id_ns)_base_link"/>

      </node>

    </group>

</launch>